libcodechecker/libhandlers/parse.py

```diff
@@ -204,8 +222,36 @@ def main(args):
                     os.chdir(metadata_dict['working_directory'])
 
             _, _, files = next(os.walk(input_path), ([], [], []))
+
             for f in files:
-                parse(os.path.join(input_path, f), context, metadata_dict,
-                      suppress_handler, 'print_steps' in args)
+                report_stats = parse(os.path.join(input_path, f),
+                                     context,
+                                     metadata_dict,
+                                     suppress_handler,
+                                     'print_steps' in args)
+
+                severity_stats.update(Counter(report_stats.get('severity',
+                                              {})))
+                file_stats.update(Counter(report_stats.get('files', {})))
+
+        if file_stats or severity_stats:
+            print("\n----==== Summary ====----")
+            vals = [[os.path.basename(k), str(v)] for k, v in
+                    sorted(dict(file_stats).items(),
+                           key=operator.itemgetter(1),
+                           reverse=True)]
+
+            keys = ['Filename', 'Report count']
+            t = twodim_to_str('table', keys, vals)
+            print(t)
+
+            vals = [[k, str(v)] for k, v in
```

Comment:
Shouldn't we extend the twodim_to_str api with sorting instead?

Reply:
I do not think the `twodim_to_str` should sort. Based on which column you want to sort might change for each use case.
We should keep it as a general table output generator.

Reply:
I mean it could have an optional parameter to sort based on a column. If no parameter is given, no sorting will be done.